control/controller.go

```diff
@@ -523,10 +529,13 @@ func (q *Query) setErr(err error) {
 // results to our destination.
 // When there are no more results, then this will close our own
 // results channel.
-func (q *Query) pump(exec flux.Query) {
+func (q *Query) pump(exec flux.Query, done <-chan struct{}) {
 	defer close(q.results)
 
-	done := q.currentCtx.Done()
+	// When the done channel is closed, we should send a signal
+	// to the underlying query to tell it to stop. We copy this
+	// to a separate channel so this signal can be done once.
+	signalCh := done
```

Comment:
I don't understand this comment. I must be missing something.  `signalCh` and `done` are always the same, so why have two different variables?

Reply:
It is modified below. `done` never changes but `signalCh` is modified to be `nil` after it receives the signal for the first time.

Reply:
Ok, now I got it.  I wasn't clear on the semantics of reading from a `nil` channel.

Reply:
Ah, ok. A nil channel will never return while a closed channel will instantly return.